for-upstream/default.nixos.nix

```
@@ -9,7 +9,7 @@ let
   defaultUser = "hercules-ci-agent";
   defaultUserDetails = {
     name = defaultUser;
-    home = "/var/lib/hercules-ci-agent";
+    home = cfg.baseDirectory;
```

I did this the other way around. I don't expect the home directory to change if I change the service baseDirectory.
This is mostly academic, unless we do the basic impure tasks, which will allow all sorts of programs to use HOME.
Perhaps we can make this a constant?
```suggestion
    home = "/var/lib/hercules-ci-agent";
```